src/app/jscaip/GamePartSlice.ts

```diff
@@ -33,10 +33,15 @@ export abstract class GamePartSlice {
     // Getters
 
     public getBoardByXY(x: number, y: number): number {
-        return this.board[y][x];
+        if (y >= 0 && y < this.board.length && x >= 0 && x < this.board[y].length) {
+            const value = this.board[y][x];
+            return value;
+        } else {

```

Comment:
`
if (0 <= y && y < this.board.length && 0 <= x && x < this.board[y].length) {
    return this.board[y][x];
}